Configuring with cmake from git results in error

Using my cmake build script
```
#!/bin/sh

find . CMakeFiles CMakeCache.txt config | xargs | rm -rf

cmake .. \
      -DCMAKE_BUILD_TYPE=Debug \
      -DCMAKE_INSTALL_PREFIX=${HOME}/local/mfem-git \
      -DBUILD_SHARED_LIBS=ON \
      -DMFEM_USE_MPI=ON \
      -DMFEM_USE_SUPERLU=ON \
      -DMFEM_USE_SUNDIALS=ON \
      -DMFEM_USE_NETCDF=ON
```
I get an error because MFEM_GIT_STRING is somehow not set.
```
mfem/general/version.cpp:61:33: error: ‘MFEM_GIT_STRING’ was not declared in this scope
    static const char *git_str = MFEM_GIT_STRING;
```
This seems to be a regression in the cmake build if I am correct.
Although it should be defined in https://github.com/mfem/mfem/blob/master/config/cmake/config.hpp.in#L34.

During the cmake execution the file `MFEMConfig.cmake` is created but with an empty expression for set(MFEM_GIT_STRING "").

I assume the problem is the interface used to access the git version which is not compatible on my git version. This is a CentOS 7 machine with git-1.8.3.1-12.el7_4.x86_64 :-1: .

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,13 +100,13 @@ mfem_version_to_int(${${PROJECT_NAME}_VERSION} ${PROJECT_NAME_UC}_VERSION)
 set(${PROJECT_NAME_UC}_VERSION_STRING ${${PROJECT_NAME}_VERSION})
 if (EXISTS ${PROJECT_SOURCE_DIR}/.git)
    execute_process(
-     COMMAND git -C ${PROJECT_SOURCE_DIR}
-                describe --all --long --abbrev=40 --dirty --always
+     COMMAND git describe --all --long --abbrev=40 --dirty --always
+     WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
      OUTPUT_VARIABLE ${PROJECT_NAME_UC}_GIT_STRING
      ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 if (NOT ${PROJECT_NAME_UC}_GIT_STRING)
-   set(${PROJECT_NAME_UC}_GIT_STRING "")
+   set(${PROJECT_NAME_UC}_GIT_STRING "(unknown)")
 endif()
 
 #-------------------------------------------------------------------------------

--- a/linalg/petsc.hpp
+++ b/linalg/petsc.hpp
@@ -181,8 +181,8 @@ public:
        PETSc format @a tid. */
    /** The supported type ids are: Operator::PETSC_MATAIJ,
        Operator::PETSC_MATIS, and Operator::PETSC_MATSHELL. */
-   PetscParMatrix(const HypreParMatrix *ha,
-                  Operator::Type tid = Operator::PETSC_MATAIJ);
+   explicit PetscParMatrix(const HypreParMatrix *ha,
+                           Operator::Type tid = Operator::PETSC_MATAIJ);
 
    /** @brief Convert an mfem::Operator into a PetscParMatrix in the given PETSc
        format @a tid. */

